main.js

```diff
@@ -1,46 +1,111 @@
-const { app, BrowserWindow, Tray, Menu } = require('electron');
-const path = require('path');
+import { app, BrowserWindow, Tray, Menu, dialog, shell } from "electron";
+import express from "express";
+import path from "path";
+import { fileURLToPath } from "url";
+import { SerialPort } from "serialport";
 
-let tray = null;
-let mainWindow = null;
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+let mainWindow, tray;
+
+const startExpressServer = () => {
+  const expressApp = express();
+
+  const staticPath = path.join(__dirname, "ElectroBlocks", "build");
+  expressApp.use(express.static(staticPath));
+
+  expressApp.get("/ports", async (req, res) => {
+    try {
+      const ports = await SerialPort.list();
+      res.json(
+        ports.map(port => ({
+          path: port.path,
+          manufacturer: port.manufacturer || "Unknown",
+          serialNumber: port.serialNumber || "N/A",
+          vendorId: port.vendorId || "N/A",
+          productId: port.productId || "N/A",
+          pnpId: port.pnpId || "N/A",
+        }))
+      );
+    } catch (error) {
+      res.status(500).json({ error: error.message });
+    }
+  });
+
+  expressApp.listen(4000, () => console.log("Server running on port 4000"));
+};
+
+const updatePortsMenu = async () => {
+  const availablePorts = await SerialPort.list();
+  const portsSubmenu = availablePorts.length

```

Comment:
So right now you have two menu systems. One for the tray app and one for a regular electron app. We just want the try app.  The other thing is that you want an endpoint for servicing electroblocks in the browser.

Reply:
We'll also want to open up port 4000 so that the user can see that being served with electroblocks